dvc/ignore.py

```diff
@@ -222,7 +240,8 @@ def _update(self, dirname):
     def _update_sub_repo(self, root, dirs):
         for d in dirs:
             if self._is_dvc_repo(root, d):
```

Comment:
Btw, looks like we might want to populate descendants with the last subrepo we've encountered. That way we will be able to distinguish uncollected path. Similar to how we do it with the ignore trie.